pkg/outdated/registry_test.go

```diff
@@ -0,0 +1,85 @@
+package outdated
+
+import (
+	"testing"
+
+	semver "github.com/hashicorp/go-version"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_SplitStringOnLen(t *testing.T) {
+	tests := []struct {
+		name              string
+		allSemverTags     []string
+		expectedOutliers  []string
+		expectedRemaining []string
+	}{
+		{
+			name: "elasticsearch",
+			allSemverTags: []string{
+				"0.0.1",
+				"0.0.2",
+				"0.8-alpha4",
+				"1.0.0",
+				"1.0.1",
+				"43.0.0",
+			},
+			expectedOutliers: []string{
+				"43.0.0",
+			},
+			expectedRemaining: []string{
+				"0.0.1",
+				"0.0.2",
+				"0.8-alpha4",
+				"1.0.0",
+				"1.0.1",
+			},
+		},
+		{
+			name: "no outliers",
```

Comment:
this does not seem like a test case with no outliers

Reply:
and in fact we don't have a no outliers test
so maybe add a test case that has major versions 1, 2, 3 and 4, with no outliers expected?

Reply:
another thing to test - 1, 2, 5, 9
Are both 5&9 detected as outliers?

Similarly - 4, 5, 6
presumably there shouldn't be any outliers there

And if we know the current semver, we should use that as the start - so that if we're on version 5, we shouldn't report 6 as an outlier just because the total set of detected major versions was 1, 2, 5, 6, 7